src/LLVM_intrinsics.jl

```diff
@@ -797,11 +798,12 @@ for (fs, c) in zip([HORZ_REDUCTION_OPS_FLOAT, HORZ_REDUCTION_OPS_INT],
 end
 
 # The fadd and fmul reductions take an initializer
-horz_reduction_version = Base.libllvm_version < v"9" ? "" : "v2."
+const horz_reduction_version = (v"9" < Base.libllvm_version < v"12") ? "v2." : ""
 for (f, neutral) in [(:fadd, "0.0"), (:fmul, "1.0")]
     f_red = Symbol("reduce_", f)
+    ff_base = Base.libllvm_version < v"12" ? "experimental.vector.reduce.$horz_reduction_version" : "vector.reduce.$horz_reduction_version"
```

Comment:
This could be declared outside the function (with a different name), next to `horz_reduction_version`.